pyvisa_py/tcpip.py

```diff
@@ -277,6 +281,35 @@ def _set_attribute(
             Return value of the library call.
 
         """
+
+        # In case of an environment with idle socket garbage collection (like docker)
+        # sockets need to be kept alive. Set pyvisa.constants.ResourceAttribute.tcpip_keepalive to enable
+        # keepalive packets even for VXI11 protocol. To read more on this issue
+        # https://tech.xing.com/a-reason-for-unexplained-connection-timeouts-on-kubernetes-docker-abd041cf7e02
+        if attribute == constants.VI_ATTR_TCPIP_KEEPALIVE:
```

Comment:
Comparing to what we have in the SOCKET part, you appear to be setting many more socket attributes, would it make sense to mirror those for SOCKET resources ?

Reply:
I believe that most of these options are needed for keepalive to fully function but I'm still not sure as I've not understood these keepalive packets fully. As I understand it, the further options tell the socket to start after one minute of being idle to send keepalive packets every minute. The first option was crucial for us as the standard was way more.

The last option is the number of retries and not really needed i guess.

I can copy them to the socket as well.